src/System.Text.Json/src/System/Text/Json/JsonEncodedText.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.Text.Json
+{
+    public readonly struct JsonEncodedText : IEquatable<JsonEncodedText>
+    {
+        private readonly byte[] _utf8Value;
+        private readonly string _value;
+
+        public ReadOnlySpan<byte> EncodedUtf8Bytes => _utf8Value;
+
+        public static JsonEncodedText Encode(string value)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+
+            return Encode(value.AsSpan());
+        }
+
+        public static JsonEncodedText Encode(ReadOnlySpan<char> value)
+        {
+            if (value.Length == 0)
+            {
+                return new JsonEncodedText(Array.Empty<byte>());
+            }
+
+            return TranscodeAndEncode(value);
+        }
+
+        private static JsonEncodedText TranscodeAndEncode(ReadOnlySpan<char> value)
+        {
+            JsonWriterHelper.ValidateValue(value);
+
+            int expectedByteCount = JsonReaderHelper.GetUtf8ByteCount(value);
+            byte[] utf8Bytes = ArrayPool<byte>.Shared.Rent(expectedByteCount);
+
+            JsonEncodedText encodedText;
+
+            try
+            {
+                // Since GetUtf8ByteCount above already throws on invalid input, the transcoding
+                // to UTF-8 is guaranteed to succeed here. Therefore, there's no need for a catch block.
+                int actualByteCount = JsonReaderHelper.GetUtf8FromText(value, utf8Bytes);
+                Debug.Assert(expectedByteCount == actualByteCount);
+
+                encodedText = EncodeHelper(utf8Bytes.AsSpan(0, actualByteCount));
+            }
+            finally
```

Comment:
If, as the comment says, there's no need for a catch block, why is there a finally?